plugin.audio.podcatcher/simplexbmc.py

```diff
@@ -0,0 +1,154 @@
+  def __init__(self,path):
+    self.path = path;
+
+  @staticmethod
+  def log(msg):
+    if not isinstance(msg,str) and not isinstance(msg,unicode):
+      xbmc.log("[%s]: %s" % (__plugin__, type(msg)))
+    else:
+      xbmc.log("[%s]: %s" % (__plugin__, msg.encode('utf8')))
+
+  def buildMediaItem(self, element, markUnread):
+    if(element.subTitle == ""):
+      title = "%s (%s)"%(element.title,element.duration);
+    else:
+      title = "%s - %s (%s)"%(element.title,element.subTitle,element.duration);
+
+    if(type(element).__name__ == 'FeedItem'):
+      title = "[%s] %s"%(time.strftime("%d.%m",element.date),title);
+
+    if(markUnread and not element.readed):
+      title = "(*) %s"%(title);
+    else:
+      title = "%s"%(title);
+    self.log(element.picture)
+    if(element.picture is not ""):
+      liz=xbmcgui.ListItem(title, iconImage="DefaultFolder.png", thumbnailImage=element.picture)
+    else :
+      liz=xbmcgui.ListItem(title, iconImage="DefaultFolder.png",)
+    liz.setInfo("music",{
+      "size": element.size,
+      "date": time.strftime("%d.%m.%Y",element.date),
+      "duration": self.durationStringToSec(element.duration),
+      "year": int(time.strftime("%Y",element.date)),
+      "artist": element.author,
+      "title": title,
+      "comment": element.description
+      });
+    return liz;
+
+  def buildMenuEntry(self, menuElement, elementCount ):
+    if(self.path == ""):
+      path = "%d"%(self.counter);
+    else:
+      path = "%s.%d"%(self.path,self.counter);
+
+    self.log(type(menuElement).__name__);
+    if(type(menuElement).__name__ == 'FeedItem'):
+      liz = self.buildMediaItem(menuElement,True);
+      liz.addContextMenuItems([(translation(4020),"XBMC.RunPlugin(%s?path=%s&action=markRead)"%(sys.argv[0],path))],True)
+      u = "%s?path=%s&action=play&guid=%s" % (sys.argv[0],path,menuElement.guid)
+      xbmcplugin.addDirectoryItem(handle=int(sys.argv[1]),url=u,listitem=liz,isFolder=True)
+
+    else:
+      if(menuElement.hasUnreadItems()):
+        title = "(*) %s"%(menuElement.title);
+      else:
+        title = "%s"%(menuElement.title);
+      self.log(menuElement.picture)
+      if(menuElement.picture is not ""):
+        liz=xbmcgui.ListItem(title, iconImage="DefaultFolder.png", thumbnailImage=menuElement.picture)
+      else :
+        liz=xbmcgui.ListItem(title, "")
+
+      contextMenuEntries = [
+        (translation(30010),"XBMC.RunPlugin(%s?path=%s&action=markRead)"%(sys.argv[0],path)),
+        (translation(30011),"XBMC.RunPlugin(%s?path=%s&action=play)"%(sys.argv[0],path)),
+        (translation(30030),"XBMC.RunPlugin(%s?path=%s&action=reload)"%(sys.argv[0],path))
+        ]
+      liz.addContextMenuItems(contextMenuEntries,True)
+      u = "%s?path=%s&action=browse" % (sys.argv[0],path)
+      xbmcplugin.addDirectoryItem(handle=int(sys.argv[1]),url=u,listitem=liz,isFolder=True,totalItems = elementCount)
+    self.counter+=1;
+
+  def play(self, playableObject):
```

Comment:
Using xbmc.Player to play files in plugins is not advised but our API doesn't allow to use setResolved URL for playlists. This might need to be improved in the future